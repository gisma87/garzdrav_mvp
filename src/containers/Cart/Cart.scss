@import "../../variables";

.Cart {
  font-family: $primary-font;
  color: $primary-color;
  display: flex;
  flex-direction: column;
  padding: 0 15px;
  @media (max-width: 800px) {
    padding: 0 0 60px;
    overflow: hidden;
    align-items: center;
  }

  &__topPanel {
    display: flex;
    justify-content: space-between;
    align-items: center;
    max-width: 800px;
    width: 100%;
    align-self: center;
    margin: 0 auto;
    padding: 0 10px 15px;
  }

  &__pageTitle {
    display: flex;
    align-items: center;
    cursor: pointer;
  }

  &__pageNumber {
    font-family: $font-number;
    display: flex;
    justify-content: center;
    align-items: center;
    width: 40px;
    height: 40px;
    border: 2px solid $primary-color;
    border-radius: 50%;
    color: $primary-color;
    font-size: 1.2rem;
    font-weight: bold;

    &_active {
      background-color: $primary-color;
      color: #fff;
    }
  }

  &__pageName {
    font-size: 1.2rem;
    margin-left: 15px;
  }

  &__pageArrow {
    font-size: 36px;
  }

  h1 {
    margin-top: 0;
    @media (max-width: 800px) {
      padding: 0 15px;
      margin-top: 20px;
      font-size: 22px;
      max-width: 450px;
      width: 100%;
      text-align: left;
    }
  }

  &__alert {
    font-size: 20px;
    padding: 10px 5px;
    background-color: #fff;
    color: darkred;
    max-width: 860px;
  }

  &__rightPanel {
    display: flex;
    flex-direction: column;
    margin: 0 auto;
    width: 100%;
    max-width: 350px;
    @media (max-width: 1200px) {
      max-width: 100%;
    }
  }

  &__containerProductsPage {
    display: flex;
    flex-direction: column;
  }

  &__mainContainer {
    display: flex;
    justify-content: space-between;
    align-items: flex-start;

    @media (max-width: 1200px) {
      flex-direction: column;
    }

  }

  &__containerInfo {
    font-size: 1.1rem;
    width: 100%;
    max-width: 820px;
    display: flex;
    flex-direction: column;
    margin-bottom: 20px;
  }

  &__infoBlock-title {
    font-size: 1.6rem;
    margin-bottom: 10px;
  }

  &__infoBlock-info {
    margin: 7px 0;
  }

  &__infoBlock-minPriceProducts {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin: 0 -20px;
    padding: 7px 20px;
    cursor: default;

    &:not(:last-child) {
      border-bottom: 1px dashed $bg-color;
    }

    &:hover {
      background-color: #FCFCEE;
    }
  }

  &__infoBlock-productTitle {
    margin-bottom: 5px;
    color: #000;
    font-size: .9rem;
    font-weight: 500;
  }

  &__infoBlock-streetRetail {
    display: flex;
    color: #000;
    font-weight: 500;
    font-size: 1rem;
  }

  &__infoBlock-brand {
    margin-right: 10px;
    color: #000;
  }

  &__infoBlock-streetText {

    margin-right: 7px;
    color: #000;
  }

  &__titlePanel {
    padding: 10px 0 15px;
    font-weight: bold;
    font-size: 1.4rem;
    color: #000;
  }

  &__promoContainer {
    margin-top: 15px;
  }

  &__promoBlock {
    border: 1px solid $primary-color !important;
    border-radius: 20px !important;
    max-width: 100% !important;
  }

  &__itemContainer {
    display: flex;
    flex-direction: column;
    flex: 1 1;
    margin-bottom: 20px;
    width: 100%;
    border: 1px solid $primary-color;
    border-radius: 15px;

    @media (min-width: 1200px) {
      margin-right: 10px;
    }

    & > div {
      border-bottom: 1px solid $primary-color;
      border-radius: 0;
    }

    & > div:first-child {
      border-top-right-radius: 15px;
      border-top-left-radius: 15px;
    }

    & > div:last-child {
      border-bottom-right-radius: 15px;
      border-bottom-left-radius: 15px;
      margin-bottom: 0;
      border-bottom: none;
    }
  }

  &__item {
    margin: 0 0 10px 0;
  }

  &__pricePanel {
    padding: 25px 30px;
    border-radius: 17px;
    display: flex;
    flex-direction: column;
    color: #000;
    background-color: $bg-color-light;
    box-shadow: $box-shadow;

    @media (min-width: 801px) and (max-width: 1200px) {
      align-items: flex-end;
      background-color: transparent;
      box-shadow: none;
    }

    @media (max-width: 800px) {
      width: 100%;
      max-width: 450px;
    }

  }

  &__retail {
    display: flex;
    flex-direction: column;
    color: #777;
  }

  &__address {
    color: $primary-color;
    font-size: 1.1rem;
    margin: 10px 0 20px;
  }

  &__warningMessage {
    font-weight: bold;
    font-size: 1.1rem;
    margin: 10px 0 20px;
  }

  &__button {
    outline: none;
    font-size: 18px;
    padding: 15px 10px;
    min-width: 182px;
    text-align: center;
    width: 100%;
    flex-basis: 45%;
    flex-grow: 1;
    margin: 5px 5px 5px 0;
    cursor: pointer;

    color: #fff;
    border-radius: 8px;

    &:hover {
      color: #fff;
      background-image: linear-gradient(to bottom, #1f5377, #1F3277);
    }
  }

  &__buttonToCart {
    min-width: 182px;
    text-align: center;
    width: 100%;
    font-size: 1.2rem;
    background: $primary-color;
    color: #fff;

    cursor: pointer;
    outline: none;
    font-weight: bold;
    padding: 15px;
    border: 1px solid $primary-color;
    border-radius: 25px;
    justify-content: center;

    &:hover {
      background-image: linear-gradient(to bottom, #1f5377, #1F3277);
    }
  }

  &__resultPrice {
    text-align: center;
    white-space: nowrap;
    font-weight: bold;
    font-size: 1.15rem;
    padding: 0 0 20px 0;
    display: flex;
    justify-content: center;
    width: 100%;

    @media (max-width: 1200px) {
      font-size: 1.3rem;
    }
  }

  &__menu {
    padding-left: 10px;
  }

  &__choiceRetail {
    padding: 0 0 20px;
    display: flex;
    flex-direction: column;
    @media (max-width: 800px) {
      padding-top: 20px;
      width: 100%;
      align-items: center;
    }

  }

  &__choiceRetailTitle {
    color: #000;
    font-size: 1.6rem;
    padding: 5px 0 20px;
  }

  &__titleChoice {
    font-size: 22px;
    font-weight: bold;
    margin: 15px 0;
    @media (max-width: 800px) {
      font-size: 16px;
      padding-left: 15px;
      max-width: 450px;
      width: 100%;
    }
  }

  &__blockMoreItems {
    max-width: 860px;
    padding: 10px 0 !important;

    @media (max-width: 800px) {
      border: none;
      max-width: 450px;
      width: 100%;
    }
  }

  &__blockTitle {
    display: flex;
    justify-content: space-between;
    max-width: 860px;
    align-items: center;
    padding-top: 20px;
  }

  //&__buttonMap {
  //  color: #000;
  //  border: 1px solid #737373;
  //  max-width: 300px;
  //}

  &__clearBtn {
    font-size: 1rem;
    cursor: pointer;

    &:hover {
      border-bottom: 1px dotted $primary-color;
      color: blue;
    }
  }

  &__menuButton {
    cursor: pointer;
    outline: none;
    font-weight: bold;
    background: #fff;
    color: $primary-color;
    padding: 8px 15px;
    border: 1px solid $primary-color;
    border-radius: 15px;
    justify-content: center;
    margin-right: 20px;

    &_active {
      background: $primary-color;
      color: #fff;
    }

    &:active {
      background: $primary-color;
      color: #fff;
    }
  }


}