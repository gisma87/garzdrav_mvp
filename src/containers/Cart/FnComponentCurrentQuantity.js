import {useEffect} from "react";

const FnComponentCurrentQuantity = (props) => {
  const {cart, cartItems, allItemRemovedFromCart} = props;

  useEffect(() => {
    if (cartItems.length < cart.length) {
      const delItems = cart.filter(item => !cartItems.some((element => element.guid === item.itemId)))
      console.log('По этим товарам сервер ответил ошибкой, они удалены из корзины: ', delItems)
      delItems.forEach(item => {
        allItemRemovedFromCart(item.itemId)
      })
    }
  })

  return null
}

export default FnComponentCurrentQuantity