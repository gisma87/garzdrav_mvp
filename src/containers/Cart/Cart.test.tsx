import {getByText, render, screen} from "@testing-library/react";
import {Provider} from "react-redux";
import store from "../../store";
import {BrowserRouter} from "react-router-dom";
import React from "react";
import Cart from "./Cart";
import {addedToCart, setCartItems} from "../../actions";
import userEvent from "@testing-library/user-event";
import '@testing-library/jest-dom'

const cartItemsTest = [
    {
        "guid": "4c3d8870-ed5b-49ec-89cd-a7e662d25e7d",
        "product": "ПИРАЦЕТАМ 200МГ. №60 ТАБ. П/П/О /ОБНОВЛЕНИЕ/",
        "manufacturer": "ОБНОВЛЕНИЕ ПФК",
        "inNameGuid": "1e8ab1cc-cc8a-4997-a898-8580c1c5ac2f",
        "inNameTitle": "Пирацетам",
        "categoryGuid": "2eef26ae-7f31-43ff-8fe5-1f17126babb5",
        "categoryTitle": "Заболевания нервной системы",
        "retails": [
            {
                "countLast": 4,
                "priceRetail": 75,
                "brand": "Дешёвая Аптека",
                "buildNumber": "34",
                "city": "Красноярск",
                "coordinates": [
                    56.046846,
                    92.946805
                ],
                "guid": "d75c02a0-2b05-40e5-b39d-1a74ac03c9e9",
                "phone": "8 (391) 299-36-36",
                "street": "Металлургов",
                "title": "Красноярск № 24",
                "weekDayTime": "08:00:00 - 22:00:00"
            },
            {
                "countLast": 3,
                "priceRetail": 76,
                "brand": "Дешёвая Аптека",
                "buildNumber": "19а",
                "city": "Красноярск",
                "coordinates": [
                    56.062851,
                    92.947065
                ],
                "guid": "68afc1b2-50df-48ae-9f5f-476ac76fe934",
                "phone": "8 (391) 216-50-77",
                "street": "Ястынская",
                "title": "Красноярск № 77",
                "weekDayTime": "08:00:00 - 22:00:00"
            },
            {
                "countLast": 1,
                "priceRetail": 75,
                "brand": "Гармония Здоровья",
                "buildNumber": "14",
                "city": "Красноярск",
                "coordinates": [
                    56.019485,
                    92.843903
                ],
                "guid": "2d12cb82-0df2-4c8d-a181-62851a7d0f94",
                "phone": "8 (391) 200-16-50",
                "street": "Железнодорожников",
                "title": "Красноярск № 50",
                "weekDayTime": "08:00:00 - 22:00:00"
            },
            {
                "countLast": 5,
                "priceRetail": 77,
                "brand": "ВСЕМ (В7)",
                "buildNumber": "24",
                "city": "Красноярск",
                "coordinates": [
                    56.038705,
                    92.872595
                ],
                "guid": "0084fb08-0468-40ae-ad04-6bb3c99399d9",
                "phone": "8 (391) 218-18-90",
                "street": "Мартынова",
                "title": "Красноярск № 64",
                "weekDayTime": "08:00:00 - 22:00:00"
            }
        ]
    },
    {
        "guid": "08a0e303-439b-4bf2-8a01-a2a4268e97e3",
        "product": "НАЗИВИН 0,01% 5МЛ. №1 НАЗАЛ.КАПЛИ ФЛ./КАП.",
        "manufacturer": "Мерк КГаА для Никомед (NIKOMED )",
        "inNameGuid": "da66fa68-4839-43af-b1c0-444df9b9fead",
        "inNameTitle": "Оксиметазолин",
        "categoryGuid": "805357b5-9d43-455d-9763-4a9eb4aead2a",
        "categoryTitle": "Насморк",
        "retails": [
            {
                "countLast": 1,
                "priceRetail": 187,
                "brand": "Дешёвая Аптека",
                "buildNumber": "8",
                "city": "Красноярск",
                "coordinates": [
                    56.032459,
                    92.775883
                ],
                "guid": "97178c64-7196-41a5-abcd-0b44bdc05ca6",
                "phone": "8 (391) 218-15-52",
                "street": "Тотмина",
                "title": "Красноярск № 52",
                "weekDayTime": "08:00:00 - 22:00:00"
            },
            {
                "countLast": 3,
                "priceRetail": 232,
                "brand": "Гармония Здоровья",
                "buildNumber": "165",
                "city": "Красноярск",
                "coordinates": [
                    55.993904,
                    92.897903
                ],
                "guid": "ddd3d84a-c559-4f29-9d29-0f4b707d9e9a",
                "phone": "8 (391) 234-45-45",
                "street": "Красноярский рабочий",
                "title": "Красноярск № 07",
                "weekDayTime": "08:00:00 - 22:00:00"
            },
            {
                "countLast": 2,
                "priceRetail": 185,
                "brand": "Дешёвая Аптека",
                "buildNumber": "34",
                "city": "Красноярск",
                "coordinates": [
                    56.046846,
                    92.946805
                ],
                "guid": "d75c02a0-2b05-40e5-b39d-1a74ac03c9e9",
                "phone": "8 (391) 299-36-36",
                "street": "Металлургов",
                "title": "Красноярск № 24",
                "weekDayTime": "08:00:00 - 22:00:00"
            }
        ]
    }
]

describe('Cart', () => {
    store.dispatch(addedToCart("4c3d8870-ed5b-49ec-89cd-a7e662d25e7d"))
    store.dispatch(addedToCart("08a0e303-439b-4bf2-8a01-a2a4268e97e3"))
    store.dispatch(setCartItems(cartItemsTest))
    const {asFragment, container} = render(
        <Provider store={store}>
            <BrowserRouter basename="/">
                <Cart/>
            </BrowserRouter>
        </Provider>
    )

    const titlePanel = container.querySelector('.Cart__titlePanel')
    const btnIncrement = container.querySelector('.CountButton__increment')
    const btnDecrement = container.querySelector('.CountButton__decrement')
    const countFirstProduct = container.querySelector('.CountButton__count')
    const priceFirstProduct = container.querySelector('.CartItem__price')
    const closeIconFirstProduct = container.querySelector('.CartItem__closeIcon')
    const dropdown = container.querySelector('.RetailsListDropdown')
    const btnDropdown = container.querySelector('.CartItem__dropdownBtn')

    test('Cart snapshot / increment / decrement', () => {
        expect(asFragment()).toMatchSnapshot()

        expect(titlePanel?.textContent).toBe('В корзине 2 товара')
        expect(countFirstProduct?.textContent).toBe('1')
        expect(priceFirstProduct?.textContent).toBe('от 75.00 ₽')
        expect(getByText(container, /ул. Мартынова 24/i)).toBeInTheDocument()

        // кликаем плюс на первом товаре 10 раз - макс.кол. - 5шт
        if (btnIncrement) {
            for (let i = 1; i < 10; i++) {
                userEvent.click(btnIncrement)
                if (i < 5) {
                    expect(countFirstProduct?.textContent).toBe(`${i + 1}`)
                    expect(priceFirstProduct?.textContent).toBe(`от ${75 * (i + 1)}.00 ₽`)
                }
            }
        }
        expect(titlePanel?.textContent).toBe('В корзине 6 товаров')
        expect(countFirstProduct?.textContent).toBe('5')
        expect(priceFirstProduct?.textContent).toBe(`от ${75 * 5}.00 ₽`)

        // кликаем минус на первом товаре 10 раз - мин.кол. - 1шт
        if (btnDecrement) {
            for (let i = 0; i < 10; i++) {
                userEvent.click(btnDecrement)
            }
        }
        expect(titlePanel?.textContent).toBe('В корзине 2 товара')
        expect(countFirstProduct?.textContent).toBe('1')

        if (dropdown && btnDropdown) {
            expect(dropdown.classList.contains('RetailsListDropdown_contentDisabled')).toBeTruthy();
            userEvent.click(btnDropdown)
            expect(dropdown.classList.contains('RetailsListDropdown_contentDisabled')).toBeFalsy();
        }

        // удалили первый продукт
        if (closeIconFirstProduct) userEvent.click(closeIconFirstProduct);
        expect(titlePanel?.textContent).toBe('В корзине 1 товар')
        expect(countFirstProduct).not.toBeInTheDocument()
    })
})


