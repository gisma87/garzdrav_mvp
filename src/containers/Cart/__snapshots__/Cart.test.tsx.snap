// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Cart snapshot / increment / decrement 1`] = `
<DocumentFragment>
  <div
    class="Cart wrapper"
  >
    <section
      class="Cart__containerProductsPage"
    >
      <div
        class="Cart__mainContainer"
      >
        <div
          class="Cart__productsContainer"
        >
          <p
            class="Cart__titlePanel"
          >
            В корзине 2 товара
          </p>
          <div
            class="Cart__itemContainer"
          >
            <div
              class="CartItem"
            >
              <svg
                class="CartItem__closeIcon"
                height="1em"
                viewBox="0 0 320 512"
                width="1em"
              >
                <path
                  d="M207.6 256l107.72-107.72c6.23-6.23 6.23-16.34 0-22.58l-25.03-25.03c-6.23-6.23-16.34-6.23-22.58 0L160 208.4 52.28 100.68c-6.23-6.23-16.34-6.23-22.58 0L4.68 125.7c-6.23 6.23-6.23 16.34 0 22.58L112.4 256 4.68 363.72c-6.23 6.23-6.23 16.34 0 22.58l25.03 25.03c6.23 6.23 16.34 6.23 22.58 0L160 303.6l107.72 107.72c6.23 6.23 16.34 6.23 22.58 0l25.03-25.03c6.23-6.23 6.23-16.34 0-22.58L207.6 256z"
                  fill="currentColor"
                />
              </svg>
              <div
                class="CartItem__container"
              >
                <div
                  class="CartItem__imageContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="CartItem__image"
                    data-icon="pills"
                    data-prefix="fas"
                    height="1em"
                    viewBox="0 0 576 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M112 32C50.1 32 0 82.1 0 144v224c0 61.9 50.1 112 112 112s112-50.1 112-112V144c0-61.9-50.1-112-112-112zm48 224H64V144c0-26.5 21.5-48 48-48s48 21.5 48 48v112zm139.7-29.7c-3.5-3.5-9.4-3.1-12.3.8-45.3 62.5-40.4 150.1 15.9 206.4 56.3 56.3 143.9 61.2 206.4 15.9 4-2.9 4.3-8.8.8-12.3L299.7 226.3zm229.8-19c-56.3-56.3-143.9-61.2-206.4-15.9-4 2.9-4.3 8.8-.8 12.3l210.8 210.8c3.5 3.5 9.4 3.1 12.3-.8 45.3-62.6 40.5-150.1-15.9-206.4z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                <div
                  class="CartItem__descriptionContainer"
                >
                  <h3
                    class="CartItem__titleLink"
                  >
                    ПИРАЦЕТАМ 200МГ. №60 ТАБ. П/П/О /ОБНОВЛЕНИЕ/
                  </h3>
                  <p
                    class="CartItem__maker"
                  >
                    ОБНОВЛЕНИЕ ПФК
                  </p>
                </div>
                <div
                  class="CartItem__priceBlock"
                >
                  <div
                    class="CountButton"
                  >
                    <button
                      class="CountButton__decrement"
                    >
                      <svg
                        height="1em"
                        viewBox="0 0 384 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M368 224H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h352c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16z"
                          fill="currentColor"
                        />
                      </svg>
                    </button>
                    <p
                      class="CountButton__count"
                    >
                      1
                    </p>
                    <button
                      class="CountButton__increment"
                    >
                      <svg
                        height="1em"
                        viewBox="0 0 384 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M368 224H224V80c0-8.84-7.16-16-16-16h-32c-8.84 0-16 7.16-16 16v144H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h144v144c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V288h144c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16z"
                          fill="currentColor"
                        />
                      </svg>
                    </button>
                  </div>
                  <p
                    class="CartItem__price"
                  >
                    от 75.00 ₽
                  </p>
                </div>
              </div>
              <div
                class="CartItem__description"
              >
                <p
                  class="CartItem__caption"
                >
                  Внешний вид товара может отличаться от изображения на сайте
                </p>
                <div
                  class="CartItem__dropdownBtn"
                >
                  <span>
                    есть в аптеках
                  </span>
                  <div
                    class="CartItem__iconContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="CartItem__arrowIcon"
                      data-icon="angle-up"
                      data-prefix="fas"
                      height="1em"
                      viewBox="0 0 320 512"
                      width="1em"
                    >
                      <path
                        d="M177 159.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 255.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 329.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class="RetailsListDropdown RetailsListDropdown_contentDisabled"
                style="height: 0px;"
              >
                <div
                  class="RetailsListDropdown__contentDropdown"
                >
                  <h3
                    class="RetailsListDropdown__title"
                  >
                    В наличии в аптеке:
                  </h3>
                  <div
                    class="RetailsListDropdown__item"
                  >
                    <p
                      class="RetailsListDropdown__titleItem"
                    >
                      ул. Железнодорожников 14
                    </p>
                    <div
                      class="RetailsListDropdown__priceContainer"
                    >
                      <p
                        class="RetailsListDropdown__count"
                      >
                        1шт. * 75 ₽/шт.
                      </p>
                      <p
                        class="RetailsListDropdown__price"
                      >
                        75.00 ₽
                      </p>
                    </div>
                  </div>
                  <div
                    class="RetailsListDropdown__item"
                  >
                    <p
                      class="RetailsListDropdown__titleItem"
                    >
                      ул. Металлургов 34
                    </p>
                    <div
                      class="RetailsListDropdown__priceContainer"
                    >
                      <p
                        class="RetailsListDropdown__count"
                      >
                        1шт. * 75 ₽/шт.
                      </p>
                      <p
                        class="RetailsListDropdown__price"
                      >
                        75.00 ₽
                      </p>
                    </div>
                  </div>
                  <div
                    class="RetailsListDropdown__item"
                  >
                    <p
                      class="RetailsListDropdown__titleItem"
                    >
                      ул. Ястынская 19а
                    </p>
                    <div
                      class="RetailsListDropdown__priceContainer"
                    >
                      <p
                        class="RetailsListDropdown__count"
                      >
                        1шт. * 76 ₽/шт.
                      </p>
                      <p
                        class="RetailsListDropdown__price"
                      >
                        76.00 ₽
                      </p>
                    </div>
                  </div>
                  <div
                    class="RetailsListDropdown__item"
                  >
                    <p
                      class="RetailsListDropdown__titleItem"
                    >
                      ул. Мартынова 24
                    </p>
                    <div
                      class="RetailsListDropdown__priceContainer"
                    >
                      <p
                        class="RetailsListDropdown__count"
                      >
                        1шт. * 77 ₽/шт.
                      </p>
                      <p
                        class="RetailsListDropdown__price"
                      >
                        77.00 ₽
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="CartItem"
            >
              <svg
                class="CartItem__closeIcon"
                height="1em"
                viewBox="0 0 320 512"
                width="1em"
              >
                <path
                  d="M207.6 256l107.72-107.72c6.23-6.23 6.23-16.34 0-22.58l-25.03-25.03c-6.23-6.23-16.34-6.23-22.58 0L160 208.4 52.28 100.68c-6.23-6.23-16.34-6.23-22.58 0L4.68 125.7c-6.23 6.23-6.23 16.34 0 22.58L112.4 256 4.68 363.72c-6.23 6.23-6.23 16.34 0 22.58l25.03 25.03c6.23 6.23 16.34 6.23 22.58 0L160 303.6l107.72 107.72c6.23 6.23 16.34 6.23 22.58 0l25.03-25.03c6.23-6.23 6.23-16.34 0-22.58L207.6 256z"
                  fill="currentColor"
                />
              </svg>
              <div
                class="CartItem__container"
              >
                <div
                  class="CartItem__imageContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="CartItem__image"
                    data-icon="pills"
                    data-prefix="fas"
                    height="1em"
                    viewBox="0 0 576 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M112 32C50.1 32 0 82.1 0 144v224c0 61.9 50.1 112 112 112s112-50.1 112-112V144c0-61.9-50.1-112-112-112zm48 224H64V144c0-26.5 21.5-48 48-48s48 21.5 48 48v112zm139.7-29.7c-3.5-3.5-9.4-3.1-12.3.8-45.3 62.5-40.4 150.1 15.9 206.4 56.3 56.3 143.9 61.2 206.4 15.9 4-2.9 4.3-8.8.8-12.3L299.7 226.3zm229.8-19c-56.3-56.3-143.9-61.2-206.4-15.9-4 2.9-4.3 8.8-.8 12.3l210.8 210.8c3.5 3.5 9.4 3.1 12.3-.8 45.3-62.6 40.5-150.1-15.9-206.4z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                <div
                  class="CartItem__descriptionContainer"
                >
                  <h3
                    class="CartItem__titleLink"
                  >
                    НАЗИВИН 0,01% 5МЛ. №1 НАЗАЛ.КАПЛИ ФЛ./КАП.
                  </h3>
                  <p
                    class="CartItem__maker"
                  >
                    Мерк КГаА для Никомед (NIKOMED )
                  </p>
                </div>
                <div
                  class="CartItem__priceBlock"
                >
                  <div
                    class="CountButton"
                  >
                    <button
                      class="CountButton__decrement"
                    >
                      <svg
                        height="1em"
                        viewBox="0 0 384 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M368 224H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h352c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16z"
                          fill="currentColor"
                        />
                      </svg>
                    </button>
                    <p
                      class="CountButton__count"
                    >
                      1
                    </p>
                    <button
                      class="CountButton__increment"
                    >
                      <svg
                        height="1em"
                        viewBox="0 0 384 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M368 224H224V80c0-8.84-7.16-16-16-16h-32c-8.84 0-16 7.16-16 16v144H16c-8.84 0-16 7.16-16 16v32c0 8.84 7.16 16 16 16h144v144c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V288h144c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16z"
                          fill="currentColor"
                        />
                      </svg>
                    </button>
                  </div>
                  <p
                    class="CartItem__price"
                  >
                    от 185.00 ₽
                  </p>
                </div>
              </div>
              <div
                class="CartItem__description"
              >
                <p
                  class="CartItem__caption"
                >
                  Внешний вид товара может отличаться от изображения на сайте
                </p>
                <div
                  class="CartItem__dropdownBtn"
                >
                  <span>
                    есть в аптеках
                  </span>
                  <div
                    class="CartItem__iconContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="CartItem__arrowIcon"
                      data-icon="angle-up"
                      data-prefix="fas"
                      height="1em"
                      viewBox="0 0 320 512"
                      width="1em"
                    >
                      <path
                        d="M177 159.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 255.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 329.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1z"
                        fill="currentColor"
                      />
                    </svg>
                  </div>
                </div>
              </div>
              <div
                class="RetailsListDropdown RetailsListDropdown_contentDisabled"
                style="height: 0px;"
              >
                <div
                  class="RetailsListDropdown__contentDropdown"
                >
                  <h3
                    class="RetailsListDropdown__title"
                  >
                    В наличии в аптеке:
                  </h3>
                  <div
                    class="RetailsListDropdown__item"
                  >
                    <p
                      class="RetailsListDropdown__titleItem"
                    >
                      ул. Металлургов 34
                    </p>
                    <div
                      class="RetailsListDropdown__priceContainer"
                    >
                      <p
                        class="RetailsListDropdown__count"
                      >
                        1шт. * 185 ₽/шт.
                      </p>
                      <p
                        class="RetailsListDropdown__price"
                      >
                        185.00 ₽
                      </p>
                    </div>
                  </div>
                  <div
                    class="RetailsListDropdown__item"
                  >
                    <p
                      class="RetailsListDropdown__titleItem"
                    >
                      ул. Тотмина 8
                    </p>
                    <div
                      class="RetailsListDropdown__priceContainer"
                    >
                      <p
                        class="RetailsListDropdown__count"
                      >
                        1шт. * 187 ₽/шт.
                      </p>
                      <p
                        class="RetailsListDropdown__price"
                      >
                        187.00 ₽
                      </p>
                    </div>
                  </div>
                  <div
                    class="RetailsListDropdown__item"
                  >
                    <p
                      class="RetailsListDropdown__titleItem"
                    >
                      ул. Красноярский рабочий 165
                    </p>
                    <div
                      class="RetailsListDropdown__priceContainer"
                    >
                      <p
                        class="RetailsListDropdown__count"
                      >
                        1шт. * 232 ₽/шт.
                      </p>
                      <p
                        class="RetailsListDropdown__price"
                      >
                        232.00 ₽
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="BlockWrapper Cart__containerInfo"
          >
            <h3
              class="Cart__infoBlock-title"
            >
              Почему указана цена "ОТ" ?
            </h3>
            <p
              class="Cart__infoBlock-text"
            >
              Цены в разных аптеках отличаются.
            </p>
            <p
              class="Cart__infoBlock-text"
            >
              Мы показываем минимальную стоимость товара в вашем городе.
            </p>
            <p
              class="Cart__infoBlock-info"
            >
              Вот список товаров из вашей корзины по минимальным ценам:
            </p>
            <div
              class="Cart__infoBlock-minPriceProducts"
            >
              <div>
                <p
                  class="Cart__infoBlock-productTitle"
                >
                  ПИРАЦЕТАМ 200МГ. №60 ТАБ. П/П/О /ОБНОВЛЕНИЕ/
                </p>
                <div
                  class="Cart__infoBlock-streetRetail"
                >
                  <p
                    class="Cart__infoBlock-brand"
                  >
                    Дешёвая Аптека
                  </p>
                  <p
                    class="Cart__infoBlock-streetText"
                  >
                    ул. Металлургов 34
                  </p>
                </div>
              </div>
              <p
                class="Cart__infoBlock-price"
              >
                <b>
                  75 ₽
                </b>
              </p>
            </div>
            <div
              class="Cart__infoBlock-minPriceProducts"
            >
              <div>
                <p
                  class="Cart__infoBlock-productTitle"
                >
                  НАЗИВИН 0,01% 5МЛ. №1 НАЗАЛ.КАПЛИ ФЛ./КАП.
                </p>
                <div
                  class="Cart__infoBlock-streetRetail"
                >
                  <p
                    class="Cart__infoBlock-brand"
                  >
                    Дешёвая Аптека
                  </p>
                  <p
                    class="Cart__infoBlock-streetText"
                  >
                    ул. Металлургов 34
                  </p>
                </div>
              </div>
              <p
                class="Cart__infoBlock-price"
              >
                <b>
                  185 ₽
                </b>
              </p>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
